.github/workflows/test.yml

```diff
@@ -0,0 +1,54 @@
+name: CI Test 
+
+# Run for any commits to any branch
+on: [push, pull_request]
```

Comment:
I'm not quite sure if with circleCi the tests where run on every branch on push or just PRs, perhaps they were but we just weren't seeing the results cause they weren't attached to a PR 🤔 . If we ever want to minimize the number of minutes we use for the projects we can always restrict it to push to main branches and pull_request: types: [opened, synchronize]

Reply:
I just saw that in the checks of the PR you have the same check twice, If we are checking on push, then it also covers when a pr is created and `pull_request` should not be needed.

Reply:
The workflow runs upon the push to the branch (on the branch code) and upon creation of the PR (on the code resulting from the merge of the branch into master). The PR checks are actually set like this automatically by GH. 
We can specify which -if any- should be required to pass in order to approve the PR.